From: Alexander Block <ablock84@googlemail.com>
Date: Sat, 28 Jul 2012 11:07:18 +0200
Patch-mainline: 3.7
Git-commit: b9291affaa4576762c30fb31083f33f5d745dea1
References: FATE#312888
Subject: [PATCH] Btrfs: add missing check for dir != tmp_dir to
 is_first_ref

We missed that check which resultet in all refs with the same name
being reported as first_ref.

Reported-by: Alex Lyakas <alex.bolshoy.btrfs@gmail.com>
Signed-off-by: Alexander Block <ablock84@googlemail.com>
Signed-off-by: David Sterba <dsterba@suse.cz>
---
 fs/btrfs/send.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/fs/btrfs/send.c
+++ b/fs/btrfs/send.c
@@ -1591,7 +1591,7 @@ static int is_first_ref(struct send_ctx
 	if (ret < 0)
 		goto out;
 
-	if (name_len != fs_path_len(tmp_name)) {
+	if (dir != tmp_dir || name_len != fs_path_len(tmp_name)) {
 		ret = 0;
 		goto out;
 	}
